### PR TITLE
Amp 44 404 page rendering

### PR DESCRIPTION
**Pull request by wajerrr**
This one should wait till amp-43 is merged as it is branched off it

@umbobabo @dosyara  @yasgreen93 just a question... my 404 is not valid amp page... :-(
because
I have removed canonical link from the page as I did not want to point to non existing Economist website article....
should I add it ? to point to this page ?

### CODE REVIEW

File: src/app/app.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import Error404 from '@economist/component-404';
+
```
Comment (umbobabo) [inline src/app/app.js]: We don't usually separate external from internal with empty lines.

File: src/server/render-html-error.test.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/* import React from 'react';
```
Comment (umbobabo) [inline src/server/render-html-error.test.js]: No commented code except pls.

File: src/server/render-html-error.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    isDev,
+  });
+};
+
+const renderHtmlError = (error, path) => {
+  switch (error.status) {
+    case 404:
+      return render404(error, path);
+    case 500:
+      return `<p>${error.toString()}</p>`;
```
Comment (umbobabo) [inline src/server/render-html-error.js]: You can save this case cause it's the same of the default.

File: src/server/render-html-error.test.js (hunk 2 above)
Comment (wajerrr) [reply src/server/render-html-error.test.js]: upss... sorry for that.  on the positive note it was easy to spot :-). removed

**Pull request merged**